webapp/graphite/finders/remote.py

```diff
@@ -240,12 +114,45 @@ def send(self, headers=None, msg_setter=None):
             }
 
             if is_leaf:
-                reader = RemoteReader(
-                    self.store, node_info, bulk_query=[
-                        self.query.pattern])
+                reader = RemoteReader(self, node_info)
                 node = LeafNode(path, reader)
             else:
                 node = BranchNode(path)
 
             node.local = False
             yield node
+
+    def fetch(self, patterns, start_time, end_time, now=None, requestContext=None):
+        reader = RemoteReader(self, {}, bulk_query=patterns)
+        return reader.fetch_multi(start_time, end_time, now, requestContext)
+
+    def request(self, url, fields=None, headers=None, timeout=None):
```

Comment:
This method centralizes the code used to make requests to remote hosts, and is also used by `RemoteReader`